Library/Homebrew/dev-cmd/bump-formula-pr.rb

```diff
@@ -107,22 +107,47 @@ def bump_formula_pr
       ]
     end
 
+    unless ARGV.dry_run?
+      backup_file = "#{formula.path}.orig"
+      if File.exist? backup_file
+        odie "Backup file #{backup_file} already exists!"
+      end
+      cp formula.path, "#{formula.path}.orig"
+    end
+
     new_contents = inreplace_pairs(formula.path, replacement_pairs)
 
     new_formula_version = formula_version(formula, requested_spec, new_contents)
 
     if new_formula_version < old_formula_version
+      mv backup_file, formula.path unless ARGV.dry_run?
       odie <<-EOS.undent
         You probably need to bump this formula manually since changing the
         version from #{old_formula_version} to #{new_formula_version} would be a downgrade.
       EOS
     elsif new_formula_version == old_formula_version
+      mv backup_file, formula.path unless ARGV.dry_run?
       odie <<-EOS.undent
         You probably need to bump this formula manually since the new version
         and old version are both #{new_formula_version}.
       EOS
     end
 
+    if ARGV.dry_run?
+      ohai "brew audit #{formula.path.basename}"
+    else
+      system HOMEBREW_BREW_FILE, "audit", "--strict", formula.path
+      if $?.success?
+        rm backup_file
+      else
+        tmp = Pathname.new(Tempfile.new("").path).basename
+        flunker = HOMEBREW_TEMP/"#{formula.name}-#{tmp}.rb"
+        mv formula.path, flunker
+        mv backup_file, formula.path
+        odie "brew audit failed! See #{flunker}"
```

Comment:
Maybe just a `git diff` of the modified formula file followed by restoring the backed up original? Would prevent one temporary file from lying around and possibly being forgotten.